components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt

```diff
@@ -40,7 +40,10 @@ class RemoteTabsFeature(
                 val lastUsed = 0L
                 // TO-DO: https://github.com/mozilla-mobile/android-components/issues/5179
                 val iconUrl = null
-                state.tabs.map { tab ->
+                state.tabs.mapNotNull { tab ->
```

Comment:
Couldn't we just stick a filter before it or did I misunderstand what we wanted to do? 🤔 

```kotlin
state.tabs.filter { !it.content.private }
```

Reply:
Yup that works too, I was wondering if the compiler would optimize the two array iterations.

Reply:
The Kotlin compiler is quite good at generating optimized bytecode, and the AOT compiler on devices do even more things to reduce runtime costs, so in general, I don't think we need to think about those kinds of optimizations on Android any more.